### PR TITLE
fix(i18n): correctly load plugin translations when caches are off

### LINKED ISSUE #11916: Plugin-provided language strings fail with caching turned off
With caching off it only outputs the keys.  With caching turned on it outputs the correct language strings.

### PR DESCRIPTION
**Pull request by hypeJunction**
Fixes #11916

### CODE REVIEW

**Comment by hypeJunction** [comment]
Translator needs a proper refactor. It's impossible to follow any logic in there, it's overly complicated for what it does.

**Review by jeabakker — APPROVED** [review]

**Comment by jeabakker** [comment]
Test error seams unrelated, but maybe because you added a new testing plugin??

File: engine/classes/Elgg/I18n/Translator.php [hunk 1]
```diff
@@ -299,6 +299,11 @@ public function loadTranslations($language = null) {
 				$this->loaded_from_cache = true;
 				$this->registerLanguagePath($this->defaultPath);
 				$this->is_initialized = true;
+
+				elgg_register_event_handler('cache:flush', 'system', function() {
```
Comment (hypeJunction) [inline engine/classes/Elgg/I18n/Translator.php]: I am not sure about this though. Does flushing caches rebuild translations from cache?

Comment (jeabakker) [reply engine/classes/Elgg/I18n/Translator.php]: it probably couldn't because the cached files are removed

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: So, my thinking is right here?

**Pull request merged**